internal/schema-manager/mysql/mysql.go

```diff
@@ -114,6 +116,20 @@ func getDatabaseDataForSchemaDiff(
 		return nil
 	})
 
+	trigmu := sync.Mutex{}
+	errgrp.Go(func() error {
+		tabletriggers, err := db.Db().GetSchemaTableTriggers(ctx, tables)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve database table triggers: %w", err)
+		}
+		for _, tabletrigger := range tabletriggers {
+			trigmu.Lock()
```

Comment:
if the mutex is actually needed, it might be better to just lock it before the for loop and then defer unlock right after. let the for loop hydrate the map and then exit, unlocking the mutex.